registrations/views.py

```diff
@@ -332,37 +334,13 @@ def post(self, request):
         serializer.is_valid(raise_exception=True)
 
         post_data = self.build_jembi_helpdesk_json(serializer.validated_data)
-        try:
-
-            source = Source.objects.get(user=self.request.user.id)
-
-            endpoint = "helpdesk"
-            if source.name == "NURSE Helpdesk App":
-                endpoint = "nc/helpdesk"
-                post_data["type"] = 12  # NC Helpdesk
-
-            result = requests.post(
-                urljoin(settings.JEMBI_BASE_URL, endpoint),
-                headers={"Content-Type": "application/json"},
-                data=json.dumps(post_data),
-                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
-                verify=False,
-            )
-            result.raise_for_status()
-        except (requests.exceptions.HTTPError,) as e:
-            if e.response.status_code == 400:
-                logger.warning(
-                    "400 Error when posting to Jembi.\n"
-                    "Response: %s\nPayload:%s"
-                    % (e.response.text, json.dumps(post_data))
-                )
-                return Response(
-                    "Error when posting to Jembi. Body: %s Payload: %r"
-                    % (e.response.content, post_data),
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-            else:
-                raise e
+        source = Source.objects.get(user=self.request.user.id)
+        endpoint = "helpdesk"
+        if source.name == "NURSE Helpdesk App":
+            endpoint = "nc/helpdesk"
+            post_data["type"] = 12  # NC Helpdesk
+        jembi_url = urljoin(settings.JEMBI_BASE_URL, endpoint)
+        request_to_jembi_api(jembi_url, post_data)
```

Comment:
This isn't something specifically for this PR, but since we're modifying this code anyway. I've always wanted this HTTP call to be done asynchronously, that way we can do retries and better error handling, and we don't block the HTTP worker from processing requests if Jembi's API is slow or down.

It should be as simple as adding a `.delay`, ie. `request_to_jembi_api.delay(jembi_url, post_data)`

This will probably mean that the one test will start failing, because we'll no longer be returning an error response if Jembi's API gives us an error (we can't since it's no longer synchronous). I'm happy to just remove that test though.

Reply:
if it affects 2 tests that I had to remove, is this still okay?

Reply:
Yes that should be fine. I'll check that it's okay once you push the two removed tests.